newm-server/src/main/kotlin/io/newm/server/features/song/repo/SongRepositoryImpl.kt

```diff
@@ -856,6 +859,47 @@ internal class SongRepositoryImpl(
         }
     }
 
+    override suspend fun getSmartLinks(songId: SongId): List<SongSmartLink> {
+        logger.debug { "getSmartLinks: songId = $songId" }
+        val minCreatedAt = LocalDateTime.now()
+            .minusSeconds(configRepository.getLong(CONFIG_KEY_SONG_SMART_LINKS_CACHE_TTL))
+        val cachedSmartLinks = transaction {
+            with(SongSmartLinkEntity.findBySongId(songId)) {
```

Comment:
Instead of select, check, then delete.  Just do a delete command with a where clause.